docs/demos/first_steps.md

```diff
@@ -0,0 +1,299 @@
+
+This installed OSM Controller in the `osm-system` namespace.
+
+## Deploy Demo Applications
+In this section we will deploy 2 pods in 2 different namespaces:
+- `bookbuyer` - a client, which makes HTTP requests to a `bookstore`
+- `bookstore` - a server, which responds to HTTP requests from `bookbuyer`
+
+
+### Create the Namespaces
+Each application in this demo will reside in a separate namespaces. Create the namespaces with:
+```bash
+kubectl create namespace bookbuyer
+kubectl create namespace bookstore
+```
+
+### Join the new namespaces to the service mesh
+The following `osm` CLI command joins the namespaces to the service mesh:
+
+`//TODO(draychev): link to relevant documentation on how the JOIN command works [#2904]`
+
+```bash
+osm namespace add bookbuyer
+osm namespace add bookstore
+```
+
+With this command each one of the four namespaces will be:
+  1. labelled with `openservicemesh.io/monitored-by: osm`
+  2. annotated with `openservicemesh.io/sidecar-injection: enabled`
+
+The OSM Controller will notice the new label and annotation on
+these namespaces and will begin injecting **new** pods with Envoy sidecars.
+
+### Create Pods
+
+Create the `bookbuyer` service account:
+```bash
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: bookbuyer
+  namespace: bookbuyer
+EOF
+```
+
+Create the `bookbuyer` pod:
+```bash
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: bookbuyer
+  name: bookbuyer
+  labels:
+    app: bookbuyer
+    version: v1
+spec:
+  serviceAccountName: bookbuyer
+  containers:
+  - name: bookbuyer
+    image: openservicemesh/bookbuyer:v0.8.0-rc.2
+    command: ["/bookbuyer"]
+    env:
+    - name: "BOOKSTORE_NAMESPACE"
+      value: bookstore
+EOF
+```
+
+Create bookstore service account:
+```bash
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: bookstore
+  namespace: bookstore
+EOF
+```
+
+Create bookstore service:
+```bash
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: bookstore
+  namespace: bookstore
+  labels:
+    app: bookstore
+spec:
+  selector:
+    app: bookstore
+  ports:
+  - port: 14001
+EOF
+```
+
+Create bookstore pod:
+```bash
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: bookstore
+  name: bookstore
+  labels:
+    app: bookstore
+spec:
+  serviceAccountName: bookstore
+  containers:
+  - name: bookstore
+    image: openservicemesh/bookstore:v0.8.0-rc.2
+    ports:
+      - containerPort: 14001
+    command: ["/bookstore"]
+    args: ["--port", "14001"]
+    env:
+    - name: BOOKWAREHOUSE_NAMESPACE
+      value: bookwarehouse
+EOF
+```
+
+To view the resources we created so far run the following commands:
+
+```bash
+kubectl get pods -A | grep -E '^book'
+
+kubectl get services -A | grep -E '^book'
+```
```

Comment:
Explicitly specify the namespace here so the commands are not specific to posix shells.